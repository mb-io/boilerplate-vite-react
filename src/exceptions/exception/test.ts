import { describe, it, expect } from 'vitest'
import { Exception } from './exception'

describe('Exception', () => {
  it('should create an exception with a default name and message', () => {
    const exception = new Exception()
    expect(exception.name).toBe('UnknownException')
    expect(exception.message).toBe('')
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with a custom message', () => {
    const message = 'Custom error message'
    const exception = new Exception(message)
    expect(exception.name).toBe('UnknownException')
    expect(exception.message).toBe(message)
    expect(exception).toMatchSnapshot()
  })
})
