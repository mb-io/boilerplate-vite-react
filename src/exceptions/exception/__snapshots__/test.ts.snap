// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Exception > should create an exception with a custom message 1`] = `[UnknownException: Custom error message]`;

exports[`Exception > should create an exception with a default name and message 1`] = `[UnknownException]`;
